### DISCUSSION TITLE
Can't use local fonts

### DISCUSSION THREAD

**Discussion by dipunj** [vercel/next.js#15841] [Help]
# Bug report

## Describe the bug

When I store my fonts at public/fonts directory, they do not work. However, I would like to point out that my images located at public/images work perfectly fine.

## To Reproduce

I have a custom ttf font which I have put in public/fonts directory and I am including it as follows in my global styles file located at styles/index.css (which has been imported into `_app.tsx` file)

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@font-face {
	font-family: 'Lobster';
	font-style: normal;
	font-display: auto;
	src: url(/fonts/Lobster-Regular.ttf) format('ttf');
}
```

and in my pages/landing.tsx file:

```tsx
const Landing = (): JSX.Element => (
            <h1 style={{ fontFamily: 'Lobster' }}>
                  Hello
            </h1>
);
export default Landing;

```
## Expected behavior

When I open localhost:3000/landing, I expect "Hello" to be rendered using Lobster font and not the default fallback fonts.

## Screenshots
<img width="1440" alt="Screenshot 2020-08-04 at 03 16 45" src="https://user-images.githubusercontent.com/23742442/89231534-2a3b5980-d603-11ea-9a41-4b79c4694665.png">
<img width="1440" alt="Screenshot 2020-08-04 at 03 16 24" src="https://user-images.githubusercontent.com/23742442/89231541-2f000d80-d603-11ea-967f-e7931d0d21bf.png">
<img width="1440" alt="Screenshot 2020-08-04 at 03 16 15" src="https://user-images.githubusercontent.com/23742442/89231543-2f98a400-d603-11ea-812c-d94dec159032.png">
<img width="1440" alt="Screenshot 2020-08-04 at 03 16 07" src="https://user-images.githubusercontent.com/23742442/89231546-30313a80-d603-11ea-8b18-593d111e147b.png">
<img width="1440" alt="Screenshot 2020-08-04 at 03 14 23" src="https://user-images.githubusercontent.com/23742442/89231548-30c9d100-d603-11ea-936f-8c24f74c81d4.png">

## System information

- macOS
- Browser: chrome, firefox,  safari
- Version of Next.js: latest (9.5.1)
- Version of Node.js: v12.16.3

## Additional context

When I use the font directly from web via google fonts i.e

```js
<link
	href="https://fonts.googleapis.com/css2?family=Lobster&display=swap"
	rel="stylesheet"
/>
```

by adding the above in `_app.tsx` file, the font loads flawlessly.

Additionally, here is my `postcss.config.js` file which I needed to customise, since I am using tailwind.

```js
module.exports = {
	plugins: [
		'postcss-flexbugs-fixes',
		[
			'postcss-preset-env',
			{
				autoprefixer: {
					flexbox: 'no-2009',
				},
				stage: 3,
				features: {
					'custom-properties': false,
				},
			},
		],
		'tailwindcss',
	],
};
```
Which is [the default postcss config which nextjs uses](https://nextjs.org/docs/advanced-features/customizing-postcss-config), and tailwindcss added to it.

Am I doing something wrong? Please enlighten. Thanks!

**Comment by Timer** [comment]
You should be referencing the font on disk:
```
url(../public/fonts/Lobster-Regular.ttf)
```

**Reply by dipunj** [reply]
Hi @Timer,

Thank you for reading through.

I tried following what the documentation says: to reference /public/folder as just /folder. But, as you suggested, I also tried the
`url(../public/fonts/Lobster-Regular.ttf)`, but to no success.

Here is an attached screenshot for reference
<img width="1440" alt="Screenshot 2020-08-04 at 13 36 35" src="https://user-images.githubusercontent.com/23742442/89269815-0e15d780-d658-11ea-9a71-a932ffc5f87e.png">

**Comment by SawkaDev** [comment]
Having the same issue! Any luck @dipunj ?

**Reply by SawkaDev** [reply]
I got mine to work by doing the following, maybe try something similar

Answer for anyone who wants to figure out how to access a file stores in the /public folder in vercel deployment. Just do this

If your file is in `/public/assets` you just get it simply by the following:

`path.join(process.cwd(),"public","assets",FILENAME);`

**Reply by emmirays** [reply]
Please can you elaborate on this a little bit?

**Comment by emmirays** [comment]
mine doesn’t seem to work either,

```javascript
const aeonik = localFont({
  src: [
    {
      path: "../../public/font/Aeonik-Regular.woff2",
      weight: "400",
      style: "normal",
    },
  ],
  variable: "--font-aeonik",
});
```

<img width="1142" alt="Screenshot 2023-12-20 at 5 15 58 PM" src="https://github.com/vercel/next.js/assets/98230614/9b4d6923-576b-4da5-a0c9-7dc7225df4ea">
